### ISSUE TITLE
Issues with Version 1.0.0

### ISSUE THREAD

**Issue by helly25** [bazel-contrib/toolchains_llvm#301]
When updating to version 1.0.0 for [github.com/helly25/mbo #23](https://github.com/helly25/mbo/commit/216c604d568fac3804839bf6ce089d732f38ff87), then I get issues on Ubuntu.

Details:

* #229: PASS
* #286: PASS
* #290
  * MacOS: PASS
  * MacOS: `--linkopt=-fuse-ld=lld` FAIL: clang: error: invalid linker name in argument '-fuse-ld=lld'
  * Ubuntu: PASS
* #292
  * MacOS: PASS
  * MacOS: `--linkopt=-fuse-ld=lld` FAIL: clang: error: invalid linker name in argument '-fuse-ld=lld'
  * Ubuntu: PASS
* #294 == v1.0.0
  * MacOS: PASS
  * Ubuntu: FAIL:
```
    external/llvm_toolchain_llvm/bin/../include/c++/v1/__config:426:14: fatal error: module 'crosstool' is needed but has not been provided, and implicit use of module files is disabled
      426 | #    include <features.h>
```
* #299
  *  MacOS: PASS
  * Ubuntu: FAIL: ditto

So #292 Fails on MacOS if `lld` is used but passes on Ubuntu and on MacOS with default linker.

Since #294 however on Ubuntu the C++20 modules appear to be active incorrectly. This can also not be disabled by passing necessary compiler flags.

**Comment by ahojnnes** [comment]
I am facing the same issue. @helly25 Any workaround that you discovered?

**Comment by helly25** [comment]
I could not find an easy solution. So for the time being I have pinned my repos to a commit hash just before version 1.0.0.

**Comment by hcoona** [comment]
`--cxxopt=-Xclang --cxxopt=-fno-cxx-modules` can mitigate it but we need a fundamental solution.

**Comment by jsharpe** [comment]
> `--cxxopt=-Xclang --cxxopt=-fno-cxx-modules` can mitigate it but we need a fundamental solution.

@fmeum's fix in https://github.com/bazel-contrib/toolchains_llvm/pull/337 should have resolved that issue - are you seeing it even with that patch applied? Appreciate it hasn't made it into a release yet, we're probably overdue cutting a new version.

**Comment by helly25** [comment]
I tried #337 with https://github.com/helly25/mbo/releases/tag/0.2.27 but I have a private repository where it causes issues. I'll investigate that further.

**Comment by helly25** [comment]
After applying the fix from #337 the mentioned private repository works fine after upgrading to Bazel 7.2.1.

@ahojnnes Feel free to reopen the bug if this still fails for you with an actual supported Bazel version.
